src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -1166,6 +1168,11 @@ public void clearActivePotionEffects() {
         }
     }
 
+    protected void updatePotionEffectsMetadata() {
+        metadata.set(MetadataIndex.POTION_COLOR, potionEffects.size() > 0 ? 1 : 0); //TODO: calculate potion particles color
```

Comment:
I believe using `Color.mixColors()`, mapping each `potionEffect.getColor()` to an array, and then calling `Color#asRGB` on the resulting mixed color will work here.

Reply:
`Color#mixColors(Color...)` is not static. I'am not sure on what color i should call it? White?

Reply:
Perhaps the first color?

Reply:
With that, it would be preferable to no longer use streams and just have to use a  i=1 -> length for loop, unfortunately.

Reply:
I retrieve base particle colors of all potion effect from bukkit with this code:
```java
for (PotionEffectType type : PotionEffectType.values()) {
    if (type != null) {
        System.out.println("type=" + type.getName() + ", color=" + type.getColor().asRGB());
    }
}
```
And then mix all colors with color of first effect of entity and result looks not like vanilla color.

Jump effect particle from glowstone:
![glowstone_jump](https://user-images.githubusercontent.com/33604982/82830569-c38d1780-9ebe-11ea-8869-dbd263c3058d.png)
Jump effect particle from vanilla:
![vanilla_jump](https://user-images.githubusercontent.com/33604982/82830571-c425ae00-9ebe-11ea-94a4-5f287f15722c.png)

Jump + strength effect particle from vanilla:
![vanilla_strength+jump](https://user-images.githubusercontent.com/33604982/82830603-e9b2b780-9ebe-11ea-87a4-099007fb8a1a.png)
Jump + strength effect particle from glowstone:
![glowstone_strength+jump](https://user-images.githubusercontent.com/33604982/82830605-ea4b4e00-9ebe-11ea-8834-7831a0187d61.png)

Reply:
I'm not sure how else to mix it, we will have to get someone to document it for us from vanilla. But for now, this can be good enough functionality with a TODO